test/src/mbgl/test/http_server.cpp

```diff
@@ -205,5 +205,81 @@ HttpServer::~HttpServer() {
     serverThread.join();
 }
 
+// Node Sever
```

Comment:
Is this `NodeHttpServer` really needed? Is it possible to run the Node server before starting the test instead? Passing the Node.js executable and forking does not look very clean.

Reply:
The NodeHttpServer just replace the original HttpSever to support all test finish.

Reply:
This implementation is from the history of mapbox-gl-native.

Reply:
I add a description at NodeHttpServer.

Reply:
In my opinion starting the test server should not be done with C++ code, especially not if the server itself is not implemented with C++. If you are able to, please refactor to remove `NodeHttpServer` and instead run it with a shell command before running the test. `fork` and the other system calls you use also not cross-platform.

We can create a ticket to remove the old HttpServer completely after this is merged since it seems redundant now.

Reply:
If you disagree, let me know. In any case I would want NodeHttpServer in a separate header / source file and to get rid of the preprocessor macros.

Reply:
Another option is using the boost beast, but it's deleted in the Mapbox boost edition. 
You know I am a rendering engineer, and I am not good at socket things, maybe the `httplib` could do some improvement to support the concurrent request unit tests, and we needn't do any changes in `mbgl-test`, but the situation now is the current `HttpSever` can't pass the unit test `HTTPFileSource.Load` and `OnlineFileSource.Load`.
But if I use the `NodeHttpServer` these unit tests can pass without any issues.

So, I could take try to use boost beast to build a new test server but it will take time, as I am not good at this area. Or we can just use the NodeHttpServer(the Mapbox legacy unit test server) to let all unit test pass and do the corresponding improvements in the future.